src/lib.rs

```diff
@@ -128,29 +128,29 @@ pub trait CollectionConfig {
     /// Configure how indexes should be created and synchronized for the associated collection.
     ///
     /// This method has a default implementation returning no index (only special `_id` index will be present).
-    fn indexes() -> index::Indexes {
-        index::Indexes::default()
+    fn indexes() -> Indexes {
+        Indexes::default()
     }
 }
 
 /// Utilities methods to get a `Repository`. Implemented for `mongodb::Database`.
 pub trait ToRepository {
     /// Shorthand for `Repository::<Model>::new`.
-    fn repository<M: Model>(&self) -> Repository<M>;
+    fn repository<M: Model + Send + Sync>(&self) -> Repository<M>;
```

Comment:
question: Do we really need `Send + Sync` bounds everywhere? Should we put these bounds directly on the `Model` trait instead?

Reply:
Good catch. I make **Model** trait **Send + Sync** and remove those bounds elsewhere and used it only where it is applicable.